include/mbedtls/ssl.h

```diff
@@ -382,6 +388,28 @@ union mbedtls_ssl_premaster_secret
 
 #define MBEDTLS_PREMASTER_SIZE     sizeof( union mbedtls_ssl_premaster_secret )
 
+static inline int mbedtls_ssl_chk_buf( uint8_t *cur, const uint8_t *end, size_t need )
+{
+    return( end < cur || (size_t)( end - cur ) < (size_t) need );
+}
+
+#define MBEDTLS_SSL_CHK_BUF_PTR( cur, end, need )			\
+    do {								\
+        if( mbedtls_ssl_chk_buf( ( cur ), ( end ), ( need ) ) != 0 )	\
+        {								\
+            return( MBEDTLS_ERR_SSL_BUFFER_TOO_SMALL );			\
+        }								\
+    } while( 0 )
```

Comment:
Some bounds checks provide a debug message when they fail, some are not; more importantly, some of the latter even _cannot_ provide a message because they happen in a context where `MBEDTLS_SSL_DEBUG_MSG` is not available (an `ssl_context *ssl` must be declared). I don't see an alternative to the current solution without changing the current behaviour.